packages/web-frontend/src/utils/periodGranularities.js

```diff
@@ -259,19 +261,40 @@ const getDefaultDatesForRangeGranularities = (periodGranularity, defaultTimePeri
   return { startDate: moment(DEFAULT_MIN_DATE), endDate: moment() };
 };
 
-export function getDefaultDates(viewConfig) {
+export function getDefaultDateRangeToFetch(viewConfig) {
   const { periodGranularity, defaultTimePeriod } = viewConfig;
 
   // we need a valid granularity to proceed
   if (!periodGranularity) {
     return {};
   }
 
+  // Fetch data for specified range and determine default dates based on that
+  const { useDatesOfData, ...restOfDefaultTimePeriod } = defaultTimePeriod || {};
+  if (useDatesOfData) {
+    return getDefaultDatesForRangeGranularities(periodGranularity, restOfDefaultTimePeriod);
```

Comment:
If using `useDatesOfData` always fetch for a range regardless of the period granularity. It's up to the report to implement logic to ensure that it only returns data for a single period.

Not ideal to have this implicit coupling between the report and the date picker behaviour, but I do feel it's somewhat unavoidable...

Reply:
I don't even see it as a requirement that the report returns data for a single period, that would depend on the type of viz being displayed. If it was a time series chart, go ahead and return all data as different periods, and it'll be fine! So in that way it's more of a coupling between the report and the display type, which is not quite as bad. 

I did think there was scope for a `usePeriodOfMostRecentData` type behaviour, where it respects a period granularity like `one_month_at_a_time` and trims all data prior to the most recent. However I see this would push the trimming logic into the report server in some way. Probably the cleanest way to do that is to define a new aggregation of `MOST_RECENT_PERIOD` which takes a config like `{ periodType: 'month' }`, and trims all data before that. That puts into the report building process but in a place that a) can be applied pretty consistently and easily to every report type that might use it, and b) can be fairly easily implemented in the db, so at least internally fetched data can have the trimming done at that level. Caveat being it would only work when the period wasn't being transformed during the report building process!

Reply:
That's a good point, the coupling isn't too bad, as the report being responsible for generating the type of data we want to display is already a bit of a requirement 👍 

As to using an aggregation vs report-server transforms to 'trim' the data, I can definitely see the benefit of this being done in aggregator, however in the specific case that I need this behaviour for it's a little bit awkward.

I need to display data for any facility that returned data in the same month as the most recent response from any facility, and I want to fetch at the country level, for multiple data elements. Our aggregations largely work independently across dimensions, whereas the filtering we'd need to do would be intentionally across all dimensions. Not saying it wouldn't be possible, but feels a bit of a breach of how aggregations are supposed to behave. However I could see the benefit of an aggregations that say 'returns all analytics in the same month as the most recent response per org unit / data element' as an independent thing 👍 

So I had actually planned on doing this in the report-server transforms instead. Was tossing up between just using an generic `excludeRows` to do this, or to add an alias transform called something like `excludeRowsNotInMostRecentMonth` (probably can come up with a better name 😛)